src/services/artifact/artifact.ts

```diff
@@ -1,10 +1,19 @@
 import * as artifact from '@actions/artifact';
+import { exec } from '@actions/exec';
 import * as fs from 'fs';
 import * as glob from '@actions/glob';
 
 class Artifact {
   async uploadLogs() {
     const basePath = '/home/runner/snap/charmcraft/common/cache/charmcraft/log';
+    // We're running some charmcraft commands as sudo as others as a
+    // regular user - we want to capture both.
+    const args = [
+      'cp',
+      '/root/snap/charmcraft/common/cache/charmcraft/log/*log',
```

Comment:
The `*` is a shell thing, it does not work as a command line argument. For `cp`, add a `.` like `/root/snap/charmcraft/common/cache/charmcraft/log/.` at the end of the dir will copy all the content of the dir, which acts like a `*`.

Also, I think it's worth checking if the src and dest dir exist before copying.